core/plugins/org.polarsys.capella.core.data.fa.ui.quickfix/src/org/polarsys/capella/core/data/fa/ui/quickfix/resolver/TC_DF_15_Resolver.java

```diff
@@ -51,11 +56,46 @@ private void addFunctionRealization(Project project, EObject obj) {
     BlockArchitecture currentLevelArchitecture = BlockArchitectureExt.getRootBlockArchitecture(obj);
     List<BlockArchitecture> previousArchitectures = BlockArchitectureExt
         .getPreviousBlockArchitectures(currentLevelArchitecture);
-    targetFunction = BlockArchitectureExt.getRootFunction(previousArchitectures.get(previousArchitectures.size() - 1));
-    sourceFunction = (AbstractFunction) obj;
-    FunctionRealization rlz = FaFactory.eINSTANCE.createFunctionRealization();
-    rlz.setSourceElement(sourceFunction);
-    rlz.setTargetElement(targetFunction);
-    sourceFunction.getOwnedFunctionRealizations().add(rlz);
+    if(!previousArchitectures.isEmpty()) {
+      AbstractFunction rootfunc = BlockArchitectureExt.getRootFunction(currentLevelArchitecture);
+      if(rootfunc.getOwnedFunctionRealizations().isEmpty()) {
+        targetFunction = BlockArchitectureExt.getRootFunction(previousArchitectures.get(previousArchitectures.size() - 1));
+        sourceFunction = (AbstractFunction) obj;
+        FunctionRealization rlz = FaFactory.eINSTANCE.createFunctionRealization();
+        rlz.setSourceElement(sourceFunction);
+        rlz.setTargetElement(targetFunction);
+        sourceFunction.getOwnedFunctionRealizations().add(rlz);
+      }
+    }
+  }
+
+  @Override
+  protected boolean enabled(Collection<IMarker> markers) {
+    for (IMarker iMarker : markers) {

```

Comment:
Some thing wrong with this return. It should check for all markers but not just the first one.